### DISCUSSION TITLE
Stop listening (close the websocket) when there is silence

### DISCUSSION THREAD

**Discussion by aawezk786** [deepgram/community#354] [General help] [answered]
Hey Deepgram Team.

if we identify silence time then will send it to stt. I want to stop listening if user pause for atleast 2 secs

**Comment by aawezk786** [comment]
@lukeocodes Can you please guide me here?

**Reply by aawezk786** [reply]
@lukeocodes any idea about above case?

**Comment by lukeocodes** [comment, marked as answer]
To stop listening after a pause, I would not use the Deepgram API for this. Instead, this would be client-side code. I suggest using MediaRecord and AudioContext to return an analyser. The browser analyser can tell you decibels. You can set minimum decibels, and time how long it is under that threshold. IF it reaches your timeout, then send a .finish() to the Deepgram websocket with the SDK.

**Reply by aawezk786** [reply]
How can i use this in backend side? @lukeocodes

**Comment by shirgoldbird** [comment]
For anyone else who might come across this question in the future, here's our [guide](https://developers.deepgram.com/docs/understand-endpointing-interim-results) explaining the difference between speech_final (which uses our Endpointing feature) and is_final (which uses our Interim Results feature).
